code/API_definitions/qos-profiles.yaml

```diff
@@ -128,14 +142,9 @@ paths:
 
 components:
   securitySchemes:
-    oAuth2ClientCredentials:
```

Comment:
I think it makes no sense to remove oAuth2ClientCredentials and replace it with openid because this is an API that is client-credentials only and API consumers should no that from this yaml file.
Please see https://github.com/camaraproject/IdentityAndConsentManagement/issues/171
There are more client-credentials only APIs now in Camara and I think TSC should re-visite their decision.

Reply:
No, it is not client credentials only. If a 3-legged access token is used, then only profiles available to the identified end user are returned. Otherwise all profiles are returned, but that does not mean that all these profiles are available to all end users.

The `openId` scheme supports all three authorisation schemes allowed by CAMARA, including the `client_credentials` grant, so only a single scheme need be specified in the OAS definition. This was the approach agreed by ICM.

Reply:
This behaviour you describe here in not described in the yaml. I assume that many API consumers are only looking at the yaml and miss this information.

Maybe it is just me, but as a developer I want the API definition to be complete. Where is the API behaviour documented?

Reply:
> If a 3-legged access token is used, then only profiles available to the identified end user are returned. Otherwise all profiles are returned, but that does not mean that all these profiles are available to all end users.

If we agree on this behaviour then I agree that it should to be described in the documentation of the API. This difference in results can't be read out of the ICM documentation.

Reply:
I propose that this is resolved for now (based on ICM v0.2.0): the mandatory documentation is in and within the operations the different behaviour with 3-legged vs 2-legged is documented.